src/map/skill.cpp

```diff
@@ -17258,26 +17252,51 @@ int32 skill_unit_onplace_timer(struct skill_unit *unit, struct block_list *bl, t
 			if( bl->id == ss->id )// it won't trigger on caster
 				break;
 			[[fallthrough]];
+		case UNT_FLASHER:
+			// Flasher doesn't activate on plant bosses
+			if (sg->unit_id == UNT_FLASHER && bl->type == BL_MOB && tstatus->class_ == CLASS_BOSS && tstatus->race == RC_PLANT)
+				break;
+			[[fallthrough]];
+		case UNT_LANDMINE:
 		case UNT_BLASTMINE:
 		case UNT_SHOCKWAVE:
 		case UNT_SANDMAN:
-		case UNT_FLASHER:
 		case UNT_FREEZINGTRAP:
 		case UNT_FIREPILLAR_ACTIVE:
 		case UNT_CLAYMORETRAP:
 		{
-			int32 bl_flag = sg->bl_flag;
-			if (tsc && tsc->getSCE(SC__MANHOLE))
+			if (tsc != nullptr && tsc->getSCE(SC__MANHOLE) != nullptr)
 				break;
+
+			int32 bl_flag = sg->bl_flag;
 			if (sg->unit_id == UNT_FIRINGTRAP || sg->unit_id == UNT_ICEBOUNDTRAP || sg->unit_id == UNT_CLAYMORETRAP)
 				bl_flag = bl_flag|BL_SKILL|~BCT_SELF;
-			map_foreachinrange(skill_trap_splash, &unit->bl, skill_get_splash(sg->skill_id, sg->skill_lv), bl_flag, &unit->bl, tick);
+
+			block_list* center = &unit->bl;
+			if (sg->unit_id == UNT_SANDMAN || sg->unit_id == UNT_FREEZINGTRAP)
```

Comment:
How about a switch including the others `sg->unit_id` above?

Reply:
Hmm I guess it would work for now as there are no overlaps, but there could be overlaps in renewal in the future and then it gets annoying.